File: src/models/Cell.js
class Cell {

    constructor(column, row) {
        this.column = column, this.row = row;
        this.calculateCellPositionOnCanvas();
        this.state = Math.random();
    }

    update() {
        let totalState = 0;
        for (let neighbor of this.neighbors) {
            //if (neighbor.cell.state >= 0.1)
            totalState += neighbor.cell.state * neighbor.kernelValue;
        }
        this.state = totalState + this.state*kernel[1][1];      
        this.cutState();
    }

    draw() {
        context.fillStyle = this.getColorForCellState(this.state);
        context.fillRect(this.column, this.row, 1,1);
        //context.fillRect(this.x, this.y, this.width, this.height);
    }

    calculateCellPositionOnCanvas() {
        this.x = canvasWidth/columnNumber*this.column;
        this.y = canvasHeight/rowNumber*this.row;
        this.width = Math.floor(canvasWidth/columnNumber);
        this.height = Math.floor(canvasHeight/rowNumber);
    }

    findNeighbors(cellMatrix) {
        this.neighbors = [];
        let currentCoordinate = new Coordinate(this.column, this.row);
        let neighborCoordinate, neighborCell;
        for (let direction in DirectionEnum) {
            neighborCoordinate = currentCoordinate.getNeighborCoordinate(DirectionEnum[direction]);
            neighborCell = cellMatrix[neighborCoordinate.column][neighborCoordinate.row]; 
            this.neighbors.push({
                kernelValue : kernel[1+DirectionEnum[direction].row][1+DirectionEnum[direction].column],
                cell : neighborCell
            })
        }
    }

    cutState() {
        if (this.state > 1) 
            this.state = 1;
        if (this.state < 0) 
            this.state = 0;
    }

    static getColorForCellState(state) {
        let alpha = (state*255).toString(16).split('.')[0];
        if (alpha.length <= 1) 
        alpha = '0' + alpha;
        return '#000000' + alpha;
    }

}

File: lib/src/main/java/NeuralCellularAutomata/model/Cell.java
package NeuralCellularAutomata.model;

import java.util.ArrayList;
import java.util.Random;
import static NeuralCellularAutomata.global.Configuration.*;

public class Cell {
    private int x, y;
    private int column, row;
    private double state;

    private ArrayList<Neighbor> neighbors;

    public Cell(int column, int row) {
        this.column = column;
        this.row = row;
        this.state = new Random().nextDouble();
        this.calculateCellPositionOnCanvas();
    }

    public void update() {
        double totalState = 0;
        for (Neighbor neighbor : this.neighbors) {
            //if (neighborCell.state > 0.1)
            totalState += neighbor.cell.state * neighbor.kernelValue;
        }
        this.state = totalState + this.state*KERNEL[1][1];
        this.state = this.state > 1 ? 1 : this.state < 0 ? 0 : this.state;
    }

    public int[][] draw(int[][] frame) {
        frame[this.column][this.row] = 0xffffff00 | (this.getColorForCellState() >> 24);
        return frame;
    }

    private void calculateCellPositionOnCanvas() {
        this.x = CANVAS_WIDTH/COLUMN_NUMBER*this.column;
        this.y = CANVAS_HEIGHT/ROW_NUMBER*this.row;
    }

    public void findNeighbors(Cell[][] cellMatrix) {
        this.neighbors = new ArrayList<Neighbor>();
        Coordinate currentCoordinate = new Coordinate(this.column, this.row);
        Coordinate neighborCoordinate;
        Cell neighborCell;
        for (Direction dir : Direction.values()) {
            neighborCoordinate = currentCoordinate.getNeighborCoordinate(dir);
            neighborCell = cellMatrix[neighborCoordinate.getColumn()][neighborCoordinate.getRow()];
            this.neighbors.add(new Neighbor(
                    KERNEL[1+ dir.getRow()][1+dir.getColumn()],
                    neighborCell
            ));
        }
    }

    private int getColorForCellState() {
        return (int) (this.state*255);
    }

    private class Neighbor {
        public double kernelValue;
        public Cell cell;
        public Neighbor(double kernelValue, Cell cell) {
            this.kernelValue = kernelValue;
            this.cell = cell;
        }
    }
}
